### PR TITLE
Feature/it hurts because its good

### PR DESCRIPTION
**Pull request by mlanca-c**
Things I did:

* Changed everything I could to smart pointers
* Changed class attribute names to start with m_ (just like duarte does it)
* Changed Config.hpp (This is the config parsing I did, I just changed my code and this doesn't affect what were doing at all)
* formatted code with .clang-format

What we have for now:
* The program passes all tests
* No leaks (obviously)
* Cleaner code

# Some observations:
* I don't think we should have `huge commented pieces of code` in our main branch. I think this is confusing to those who did not comment and don't really understand what its doing there. Also cleaner files in main which is a plus.
* `clang-format is apparently not formatting code you feel is readable`. I always format the code before making any pull request and if I feel something needs to be changed in .clang-format I change it, (you could do the same as well, is just a 5 minute google search), making sure your code is formatted before delivering it actually saves me a lot of time. :) and believe me I've already spent a lot of time formatting your code.
* I for one think clang-format is a good feature to have (cause we're 3 and we have different code styles and clang-format could be a mix of them all and bring some good consistency to our work). That being said, if you guys don't feel like doing this, we could always just give up on formatting.

### CODE REVIEW

File: tests/test_HTTPParser.cpp [hunk 1]
```diff
@@ -144,12 +143,13 @@ TEST_F(test_HTTPParser, parse_header) {
     EXPECT_STREQ(res2->getHeader("Accept-Encoding").c_str(), " gzip, deflate");
     ASSERT_FALSE(res2->getHeader("Connection").empty());
     EXPECT_STREQ(res2->getHeader("Connection").c_str(), " Keep-Alive");
-
 }
 
 // TEST_F(test_SocketConnection, hardcore) {
 //     ASSERT_NE(this->connection, nullptr);
-//     this->client->send_message("GET /usr/home HTTP/1.1\r\nHost:x\r\nContent-Length: 120\r\n\r\n........................................................................................................................"); 
+//     this->client->send_message("GET /usr/home
```
Comment (Dutesier) [inline tests/test_HTTPParser.cpp:150]: Nope, this will not compile once uncommented

**Review by Dutesier — CHANGES_REQUESTED** [review]
clang-format is a double edge sword :)

File: srcs/utils/utils.cpp [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+std::string utils::d_path(std::string p) {
+
+    static std::string path = "../webserv/";
+    if (!p.empty()) { path = p; }
+    return (path);
+}
+
+std::string utils::d_root(std::string r) {
+
+    static std::string root = d_path() + "website/";
```
[the remark this answers is not in the document]
Comment (J0Santos) [reply srcs/utils/utils.cpp:21]: https://stackoverflow.com/questions/1005685/c-static-initialization-order

Nope, no guarantee...if we want to do this, it seems we have to wrap it in a function...

[the remark this answers is not in the document]
Comment (J0Santos) [reply srcs/utils/utils.cpp:21]: but this is for static vars in 2 different files, not sure if it works on when on the same file

[the remark this answers is not in the document]
Comment (J0Santos) [reply srcs/utils/utils.cpp:21]: https://stackoverflow.com/questions/38148438/c-static-initialization-order-fiasco-happens-when

Its not a problem

**Review by Dutesier — APPROVED** [review]

**Pull request merged**